app/src/test/java/com/example/trainingintive/dog_images_feature/domain/usecase/GetDogImageUrlUseCaseTest.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.trainingintive.dog_images_feature.domain.usecase
+
+import com.example.trainingintive.dog_images_feature.domain.model.DogImageUrl
+import com.example.trainingintive.dog_images_feature.domain.repository.DogImageRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Single
+import org.junit.Test
+
+class GetDogImageUrlUseCaseTest {
+
+    val dogImageUrl =
+        DogImageUrl("https:\\/\\/images.dog.ceo\\/breeds\\/bulldog-french\\/n02108915_9666.jpg")
+    val repository: DogImageRepository = mockk()
+    val tested = GetDogImageUrlUseCase(repository)
+
+    @Test
+    fun execute() {
+        every { repository.getDogImageUrl() } returns Single.fromCallable { dogImageUrl }
```

Comment:
Minor: You can use `Single.just(dogImageUrl)` for simplicity. Also, in this case `Single.never()` can be probably used to get rid of `dogImageUrl` from this test because you are only testing if `repository.getDogImageUrl()` is called, so result is not important for you in this case.

Reply:
BUT please also add tests which will verify value returned from repo (test if use case is returning value from repo) apply it also to other tests